src/components/OmniSearch/OmniSearch.jsx

```diff
@@ -0,0 +1,180 @@
+import React, { PureComponent, PropTypes } from 'react';
+import Autosuggest from 'react-autosuggest';
+import d3 from 'd3';
+
+import './OmniSearch.scss';
+
+/**
+ * Omni Search component.
+ * Allows for auto completing Location Searches
+ */
+export default class OmniSearch extends PureComponent {
+
+  static propTypes = {
+    onSearchChange: PropTypes.func,
+    searchQuery: PropTypes.string,
+    searchResults: PropTypes.array,
+  }
+
+  static defaultProps = {
+    searchQuery: '',
+    searchResults: [],
+  }
+
+  static contextTypes = {
+    router: PropTypes.object,
+  }
```

Comment:
New preferred method is to use react-router's `withRouter()` around the component export. See https://github.com/reactjs/react-router/blob/master/docs/API.md#withroutercomponent-options.

It makes `router` a prop. You'd do something like:

```js
import { withRouter } from 'react-router';
class OmniSearch extends PureComponent { ... }
export default withRouter(OmniSearch)
```

And then instead of a contextType, `router` becomes a normal prop.